### PR TITLE
Tinyinst Update

### PR DESCRIPTION
**Pull request by elbiazo**
Initial code to get tinyinst working on LibAFL with test code in `./fuzzer/tinyinst_simpole`.

It is still really rough and is really really slow. I think the instrumentation->Run() function is the cause but not sure. I think it would be also better to 

1. add new MapObserver to match Jackalope (https://twitter.com/ifsecure/status/1577814867746213888?t=).
2. better way to build TinyInst with correct value instead of tieing to my repo. Maybe use diff but not sure
3. Better API to handle Tinyinst argument
4. optimization and correctly implementing tinyinst. (probably should look into jackalope code)

### CODE REVIEW

**Comment by elbiazo** [comment]
I just realized that it added alot of participant because I pulled the code from master. I am sorry 😭

File: libafl_tinyinst/src/shim.cc [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+std::unique_ptr<LiteCov> litecov_new() {
+  return std::make_unique<LiteCov>();
+}
+
+std::unique_ptr<Coverage> coverage_new() {
+  return std::make_unique<Coverage>();
+}
+
+void get_coverage_map(uint8_t *bitmap, size_t map_size, Coverage &newcoverage) {
+  for (auto iter = newcoverage.begin(); iter != newcoverage.end(); iter++) {
```
Comment (domenukk) [inline libafl_tinyinst/src/shim.cc:12]: What jackalope does is it ignores coverage for branches that have already been seen before.
That way, `newcoverage` is only filled when we have executed at least one new block.
So actually we can just (somehow? Rust<>CPP interop?) expose the list of new coverage to LibAFL, in something like a `NewCoverageObserver`, and then have a `NewCoverageFeedback` that always returns true if the count is at least one.

**Comment by domenukk** [comment]
I'll merge this to the branch immediately so it's easier to see the differences to main, but see my comment above

**Comment by domenukk** [comment]
@elbiazo I tried to cleanly merge main into tinyinst but may have broken something (think the submodule is gone, for example), please pull from the branch and see, sorry :D

**Comment by tokatoka** [comment]
super nice!

File: libafl_tinyinst/src/shim.cc (hunk 1 above)
Comment (elbiazo) [reply libafl_tinyinst/src/shim.cc:12]: oh ya i think what they do is have two coverage. one for overall bb and other for new coverage then they call merge function to merge it.

**Pull request merged**